ext/reflection/php_reflection.c

```diff
@@ -5904,29 +5873,8 @@ ZEND_METHOD(ReflectionProperty, setRawValue)
 		RETURN_THROWS();
 	}
 
-	uint32_t *guard = NULL;
-	uint32_t guard_backup;
-	zend_internal_function func_copy;
-	zend_function *func_backup = NULL;
-	if (Z_OBJCE_P(object)->ce_flags & ZEND_ACC_USE_GUARDS) {
-		guard = zend_get_property_guard(Z_OBJ_P(object), ref->unmangled_name);
-		guard_backup = *guard;
-		*guard |= ZEND_GUARD_PROPERTY_HOOK;
-
-		func_backup = EX(func);
-		func_copy = EX(func)->internal_function;
-		func_copy.prop_info = ref->prop;
-		EX(func) = (zend_function *) &func_copy;
-	}
-
-	zend_update_property_ex(intern->ce, Z_OBJ_P(object), ref->unmangled_name, value);
-
-	if (guard) {
-		*guard = guard_backup;
-	}
-	if (func_backup) {
-		EX(func) = func_backup;
-	}
+	zend_function *func = zend_get_property_hook_trampoline(ref->prop, ZEND_PROPERTY_HOOK_SET, ref->unmangled_name);
+	zend_call_known_instance_method_with_1_params(func, Z_OBJ_P(object), NULL, value);
```

Comment:
Nice!

Can we avoid using the trampoline at all when the property is not hooked?

Reply:
We can, that would be better.